Gordon360/Controllers/LostAndFoundController.cs

```diff
@@ -36,74 +36,72 @@ public ActionResult<int> CreateMissingItemReport([FromBody] MissingItemReportVie
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            await lostAndFoundService.UpdateMissingItemReportAsync(missingItemId, MissingItemDetails, authenticatedUserUsername);
 
             return Ok();
         }
 
         /// <summary>
         /// Update the status of the item report with given id to the given status text
         /// </summary>
-        /// <param name="id">The id of the report to update</param>
+        /// <param name="missingItemId">The id of the report to update</param>
         /// <param name="status"></param>
         /// <returns>ObjectResult - the http status code result of the action</returns>
         [HttpPut]
-        [Route("missingitem/{id}/{status}")]
-        public async Task<ActionResult> UpdateReportStatus(int id, string status)
+        [Route("missingitems/{missingItemId}/{status}")]
+        public async Task<ActionResult> UpdateReportStatus(int missingItemId, string status)
         {
-            await lostAndFoundService.UpdateReportStatusAsync(id, status);
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            await lostAndFoundService.UpdateReportStatusAsync(missingItemId, status, authenticatedUserUsername);
 
             return Ok();
         }
 
         /// <summary>
-        /// Get all missing item reports, only for data entry level users
+        /// Get the list of missing item reports for the currently authenticated user.
         /// </summary>
+        /// <param name="user">Query parameter, default is null and route will get all missing items, or if user is set
+        /// route will get missing items for the authenticated user</param>
         /// <returns>ObjectResult - an http status code, with an array of MissingItem objects in the body </returns>
         [HttpGet]
-        [Route("missingitemsall")]
-        [StateYourBusiness(operation = Static.Names.Operation.READ_ALL, resource = Resource.LOST_AND_FOUND_MISSING_REPORT)]
-        public ActionResult<IEnumerable<MissingItemReportViewModel>> GetAllMissingItems()
+        [Route("missingitems")]
+        public ActionResult<IEnumerable<MissingItemReportViewModel>> GetMissingItems(string user = null)
         {
-            IEnumerable<MissingItemReportViewModel> result = lostAndFoundService.GetMissingItemsAll();
-            if (result != null)
+            IEnumerable<MissingItemReportViewModel> result;
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            // If no username specified in the query, get all items
+            if (user == null)
             {
-                return Ok(result);
+                result = lostAndFoundService.GetMissingItemsAll(authenticatedUserUsername);
             }
             else
             {
-                return NotFound();
+                result = lostAndFoundService.GetMissingItems(authenticatedUserUsername);
             }
```

Comment:
Interestingly, you're never actually using the value of `user` here except to check for if it's present. That is, because you're passing `authenticatedUserUsername` to `GetMissingItems`, instead of `user`, the user will only ever get their own missing item reports, regardless of which username they pass to the API.

I think that this is essentially the correct behavior in terms of privacy/security. A user shouldn't be able to view another user's missing item reports unless they're an admin (I assume). However, that's essentially an undocumented quirk of the current code. It's easy to imagine a future developer editing this code and noticing that you never use the `user` parameter, without understanding why, and accidentally creating a privacy violation.

A more thorough approach would be to allow users to look up any other user's missing item reports, but ONLY if they actually have authorization to (i.e. are admins or assistors). Then you explicitly handle the case where e.g. the user John.Doe calls the route `GET /lostandfound/missingitems?user=Matthew.Jones` when he isn't authorized to, by checking his authorization and returning 404[^1] intentionally, rather than ignoring his invalid request and returning the missing item reports for `john.doe`.

[^1]: Cited from the [404 Not Found Spec](https://www.rfc-editor.org/rfc/rfc7231#section-6.5.4), "The 404 (Not Found) status code indicates that the origin server did not find a current representation for the target resource **_or is not willing to disclose that one exists_**." (emphasis mine).